src/calendar/export/CalendarParser.ts

```diff
@@ -873,89 +871,52 @@ type WeekDuration = {
 	type: "week"
 	week: number
 }
-const durationTimeParser: Parser<TimeDuration> = mapParser(
-	combineParsers(makeCharacterParser("T"), makeEitherParser(hourDurationParser, makeEitherParser(minuteDurationParser, secondDurationParser))),
-	([t, value]) => {
-		let minuteTuple, secondTuple
-		let hour, minute, second
-
-		if (value[1] === "H") {
-			hour = value[0]
-			minuteTuple = downcast(value)[2]
-		} else if (value[1] === "M") {
-			minuteTuple = value
-		} else if (value[1] === "S") {
-			secondTuple = value
-		}
+const durationTimeParser = mapParser(
+	combineParsers(makeCharacterParser("T"), maybeParse(hourDurationParser), maybeParse(minuteDurationParser), maybeParse(secondDurationParser)),
```

Comment:
the way it's defined now it will consume `T` and then if the rest if garbage it will happily only eat `T`. That could be a problem but I've noticed that `durationParser` is anyway defined as

```ts
	combineParsers(
		maybeParse(makeEitherParser(makeCharacterParser("+"), makeCharacterParser("-"))),
		makeCharacterParser("P"),
		maybeParse(makeEitherParser(durationDateParser, makeEitherParser(durationTimeParser, durationWeekParser))),
	),
```

which is, strictly speaking, not

```
       dur-value  = (["+"] / "-") "P" (dur-date / dur-time / dur-week)

       dur-date   = dur-day [dur-time]
       dur-time   = "T" (dur-hour / dur-minute / dur-second)
       dur-week   = 1*DIGIT "W"
       dur-hour   = 1*DIGIT "H" [dur-minute]
       dur-minute = 1*DIGIT "M" [dur-second]
       dur-second = 1*DIGIT "S"
       dur-day    = 1*DIGIT "D"
```

but in `parseDuration`  we check that we consumed the whole value so it should be fine?

We should probably define more of them in a way that reads like grammar but I think it's actually okay if you leave it like it is.